### PR TITLE
issue 459 Gocean grid property scalars are passed by value in OpenCL

### PR DESCRIPTION
**Pull request by sergisiso**
Also removes remaining 'nx' from the OpenCL set argument subroutine and fixes the indentation of a couple of Fortran lines.

### CODE REVIEW

**Comment by sergisiso** [comment]
@arporter This small PR solves issue 459 and cleans the (now) unused 'nx' argument from the OpenCL PSy layer. I tested nemolite2d with this branch to make sure everything still works. If the unit-test passes this is ready for review.

File: src/psyclone/gocean1p0.py [hunk 1]
```diff
@@ -1132,21 +1133,29 @@ def gen_arg_setter_code(self, parent):
...
+        args = [x for x in grid_prop_args if x.is_scalar()]
+        if args:
+            sub.add(DeclGen(sub, datatype="integer", intent="in",
                             target=True,
                             entity_decls=[arg.name for arg in args]))
-        # Scalars
-        args = args_filter(self._arguments.args,
-                           arg_types=["scalar"],
+
+        # Scalars arguments
```
Comment (arporter) [inline src/psyclone/gocean1p0.py]: s/Scalars/Scalar/

File: src/psyclone/gocean1p0.py [hunk 2]
```diff
@@ -1185,7 +1194,8 @@ def gen_data_on_ocl_device(self, parent):
                           funcnames=["create_rw_buffer"]))
         parent.add(CommentGen(parent, " Ensure field data is on device"))
         for arg in self._arguments.args:
-            if arg.type == "field" or arg.type == "grid_property":
+            if (arg.type == "field" or
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply src/psyclone/gocean1p0.py]: Fixed

File: src/psyclone/gocean1p0.py [hunk 3]
```diff
@@ -1245,9 +1255,9 @@ def gen_data_on_ocl_device(self, parent):
                         rhs=".true."))
 
                 # Ensure data copies have finished
-                ifthen.add(CommentGen(parent,
+                ifthen.add(CommentGen(ifthen,
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply src/psyclone/gocean1p0.py]: Fixed

File: src/psyclone/gocean1p0.py [hunk 4]
```diff
@@ -1116,9 +1119,7 @@ def gen_arg_setter_code(self, parent):
         # the OpenCL transformation.
```
[the remark this answers is not in the document]
Comment (sergisiso) [reply src/psyclone/gocean1p0.py]: Fixed both

File: src/psyclone/gocean1p0.py (hunk 1 above)
Comment (sergisiso) [reply src/psyclone/gocean1p0.py]: Fixed

**Comment by sergisiso** [comment]
@arporter This is ready for next review

**Comment by sergisiso** [comment]
@arporter I removed the Ready for Review label as I want to bring it to master and update the dl_esm_inf with the additional OpenCL support.

**Comment by sergisiso** [comment]
@arporter It is ready for review again, in addition to address the comments I updated the dl_esm_inf submodule and made some modifications regarding how the testing infrastructure builds it.

The unit-test (with compilation) passes, but codecov coverage has failed due to new non covered testing code.

**Pull request merged**